packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -723,7 +726,13 @@ export function isInterleavedUpdate(fiber: Fiber, lane: Lane) {
     // TODO: Optimize slightly by comparing to root that fiber belongs to.
     // Requires some refactoring. Not a big deal though since it's rare for
     // concurrent apps to have more than a single root.
-    workInProgressRoot !== null &&
+    (workInProgressRoot !== null ||
+      // If the interleaved updates queue hasn't been cleared yet, then
+      // we should treat this as an interleaved update, too. This is also a
+      // defensive coding measure in case a new update comes in between when
+      // rendering has finished and when the interleaved updates are transferred
+      // to the main queue.
+      hasInterleavedUpdates() !== null) &&
```

Comment:
Doesn’t this return a Boolean? And so !== null is always true?

Reply:
Yeah I fixed it on main. Made a mistake when changing the type.

It happens to work like this anyway, too. In fact it would work if this whole function returned `true`. We should consider actually doing that.